### PR TITLE
Adding loading indicator when performing password reset

### PR DESCRIPTION
**Pull request by jiashuChen**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

**Comment by jiashuChen** [comment]
> Thanks for the contribution @jiashuChen.
> 
> It looks like `validate loading indicator is triggered on form submission` test is not passing on Travis: https://app.travis-ci.com/github/okta/okta-signin-widget/jobs/557430966#L1942. Would you be able to investigate/fix it?

Hi @oleksandrpravosudko-okta,

Thanks, I've fixed the failing test and also added one extra test cases to cover successful form submission. Your review would be appreciated.

Regards,
Jiashu

File: jest.config.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ module.exports = {
   transform: {
     '^.+\\.(js)$': 'babel-jest',
   },
-  transformIgnorePatterns: ['/node_modules/', '/packages/', '/dist/', '/build/'],
+  // transformIgnorePatterns: ['/node_modules/', '/packages/', '/dist/', '/build/'],
```
Comment (aarongranick-okta) [inline jest.config.js:18]: Try this instead: 
```
transformIgnorePatterns: ['/node_modules/', '/packages/', '/dist/', '/okta-auth-js/build/']
```

**Review by aarongranick-okta — APPROVED** [review]
lgtm. someone from OKta will take care of merging this in for the next feature release (end of Feb)

File: jest.config.js (hunk 1 above)
Comment (aarongranick-okta) [reply jest.config.js:18]: This will be fixed in a separate PR

**Comment by jiashuChen** [comment]
> lgtm. someone from OKta will take care of merging this in for the next feature release (end of Feb)

hi @aarongranick-okta any update on the status of this pull request?

**Comment by aarongranick-okta** [comment]
@jiashuChen Unfortunately these changes have not been merged due to some breakages in downstream E2E tests. The team which owns this feature has taken on these PRs and will determine which parts are able to be merged.

I can suggest a workaround to get your changes into production for your org: since the widget is open source, it is possible to host your own custom-built version of the widget with all desired customizations included.  This will obviously work for an embedded widget, but I believe you can even do this for a redirect flow by setting up the okta-hosted custom domain feature and then editing the signin page template to serve your custom widget assets.

**Comment by jiashuChen** [comment]
> @jiashuChen Unfortunately these changes have not been merged due to some breakages in downstream E2E tests. The team which owns this feature has taken on these PRs and will determine which parts are able to be merged.
> 
> I can suggest a workaround to get your changes into production for your org: since the widget is open source, it is possible to host your own custom-built version of the widget with all desired customizations included. This will obviously work for an embedded widget, but I believe you can even do this for a redirect flow by setting up the okta-hosted custom domain feature and then editing the signin page template to serve your custom widget assets.

Thank you

**Pull request closed without merging**